internal/versions/versions.go

```diff
@@ -307,32 +307,41 @@ func (h *handler) GetLatestOsImage(cpuArchitecture string) (*models.OsImage, err
 	return latest, nil
 }
 
-func (h *handler) AddReleaseImage(releaseImageUrl, pullSecret, ocpReleaseVersion, cpuArchitecture string) (*models.ReleaseImage, error) {
+func (h *handler) AddReleaseImage(releaseImageUrl, pullSecret, ocpReleaseVersion string, cpuArchitectures []string) (*models.ReleaseImage, error) {
 	var err error
-	// If release version or cpu architecture are not specified, use oc to fetch values
-	if ocpReleaseVersion == "" || cpuArchitecture == "" {
+	var cpuArchitecture string
+	var osImage *models.OsImage
+
+	// If release version or cpu architectures are not specified, use oc to fetch values
```

Comment:
If the arch is specified, what happens if the arch list is wrong?

There are a few cases, but the most obvious is someone provides an arch that actually isn't present in the multiarch image.
I also see there's some handling around `amd64` vs `x86_64`, do we care about handling that at a later stage if we aren't detecting it? For example, can a user specify either in the release image arch?

I guess the same conversation could be had around the single arch image though ... :man_shrugging:

Reply:
Right, so in the current logic with single-arch if you lied and provided a wrong valued we just let it be. I copy the same behaviour. But I think it's important to note here that `AddReleaseImage()` is a function used only internally and never interfaces the user. User talks to us via `ClusterImageSet` in which my expectation (correct me if I'm wrong) is that you only give us URL of the release image and we do all the magic with detection for you.

Reply:
The same applies to `amd64` vs `x86_64` - whatever the rule was now, the same rule will apply tomorrow. Again, I would personally prefer that we just drop this logic "if function is called with values pre-filled just trust them" but I don't want to make too many changes in a single PR.

It was meant to allow passing `[]string` instead `string` so as much as the question "what if the value passed is incorrect" is a valid one, I see it out of scope of the change

Reply:
:+1: yeah, I agree this isn't something you are changing here.
Just something I noticed while reading through.

Maybe the choice in the future is just to pull the image and detect the arch ourselves every time :man_shrugging:

Reply:
Yes, I'll be very much in favour of just doing `oc` and/or `skopeo` ourselves instead of relying on what's passed. I'll definitely do it in a subsequent PR in this story, just want to get this one properly reviewed so it does not grow every day :)